onnxruntime/test/python/onnxruntime_test_python_autoep.py

```diff
@@ -100,8 +101,83 @@ def test_cuda_prefer_gpu_and_inference(self):
         output_expected = np.array([[1.0, 4.0], [9.0, 16.0], [25.0, 36.0]], dtype=np.float32)
         np.testing.assert_allclose(output_expected, res[0], rtol=1e-05, atol=1e-08)
 
-        # TODO(adrianlizarraga): Unregistering CUDA EP library causes issues. Investigate.
-        # self.unregister_execution_provider_library(ep_registration_name)
+        del sess  # Delete session before unregistering library
+        self.unregister_execution_provider_library(ep_registration_name)
+
+    def test_cuda_ep_selection_delegate_and_inference(self):
+        """
+        Test selecting CUDA EP via the custom EP selection delegate function and then run inference.
+        """
+        ep_lib_path = "onnxruntime_providers_cuda.dll"
+        ep_registration_name = "CUDAExecutionProvider"
```

Comment:
nit: the registration name and EP names don't need to match. could we call this 'ep_name' to avoid potentially creating an assumption that they always do?

Reply:
Good call. Will address in a separate PR as I actually used `ep_registration_name` for all tests. I hope it's ok to merge now to get it into main.